upgrade to gatsby v3 including all related deps

jbolda opened the pull request:
## Motivation

With an impending design update, it is time to close #135. This PR does this upgrade including most of the other dependencies (since much of this is very tied together).

## Approach

I upgraded all of the deps and worked through all of the errors.

### Possible Drawbacks or Risks

The URLs changing and the design changing with this upgrade are the biggest risks. With our previous work, I think we can be pretty confident around the former. The latter will eventually be replaced so I imagine very minor differences may not matter much.

### TODOs and Open Questions

- [x] upgrade gatsby and all deps with a build that finishes
- [x] confirm that all plugins that add functionality (especially around markdown) have migrated correctly; these tend to not actually throw errors so it is less obvious when they break
- [x] fix all errors or have a plan to eventually resolve them (like errors related to styles may not make sense to fix)

jbolda committed:
upgrade to gatsby v3 including all related deps

jorgelainfiesta reviewed (approved):
Looks good! the URLs seem to be fine, and indeed style changes are probably not important. I didn't do a very detailed inspection but I couldn't notice any difference with what's currently in production.

jbolda committed:
add note about image paths and outdated package

jbolda committed:
adjust podcast CTA input as readOnly

jbolda committed:
pass in and podcast data and use on blog list

we needed the author slug to remove some undefined errors, and it starts to address one of the open issues

jbolda committed:
add keys to blog posts

jbolda committed:
adjust config for md based images

jbolda committed:
fix hamburger menu for non webflow pages

jbolda commented:
It looks like the last remaining errors are with the styles. Our hamburger menu button has been broken on the gatsby part of the site for some time. That is fixed as well. I think this is all set, but I'm going to finish up the PR with BigTest first and get that merged in. It will give us a good before/after the upgrade test.
